frontend/src/App.js

```diff
@@ -12,46 +11,142 @@ import FileState from './context/file/FileState';
 import Login from './components/Login/Login';
 import Register from './components/Login/Register';
 import AuthContext from './context/auth/AuthContext';
+import setAuthToken from './utils/setAuthToken';
+import PrivateRoute from './utils/PrivateRoute';
+import './App.css';
 
 /** APP LOGIC EXPLAINED:
- * Basically, the application is divided into two depending if user is signed in or not:
- *  1. Signin and register pages for those who are NOT AUTHENTICATED
- *  2. Application pages for those who are AUTHENTICATED.
+ * Basically:
+ *  1. Signin and register pages redirect to home when AUTHENTICATED
+ *  2. Application pages redirect to '/' when NOT AUTHENTICATED.
+ *
+ *
  */
 
+// this is utility function for axios requests (sets token into headers).
+if (localStorage.token) {
+  setAuthToken(localStorage.token);
+}
+
 const App = () => {
   const authContext = useContext(AuthContext);
   const { isAuthenticated } = authContext;
   console.log('isAuthenticated says: ', isAuthenticated);
 
-  return isAuthenticated ? (
+  return (
     <FileState>
       <div className='App-container'>
-        <div className='Navbar-container'>
-          <Navbar></Navbar>
-        </div>
+        {isAuthenticated ? (
+          <div className='Navbar-container'>
+            <Navbar></Navbar>
+          </div>
+        ) : (
+          ''
+        )}
         <div className='Mother-container'>
           <Header></Header>
           <div className='Main'>
             <Switch>
-              <Route exact path='/home' component={HomePage}></Route>
-              <Route exact path='/printers' component={PrintersPage}></Route>
-              <Route exact path='/settings' component={SettingsPage}></Route>
-              <Route exact path='/history' component={HistoryPage}></Route>
+              <Route
```

Comment:
This is pretty heavy for the brains. Maybe this could be refactored.

E.g. different returns if authenticated..